engine/src/QuadTree.h

```diff
@@ -0,0 +1,78 @@
+//
+// Created by tdancois on 7/31/16.
+//
+
+#ifndef ARES_QUADTREE_H
+#define ARES_QUADTREE_H
+
+
+#include <SFML/Graphics/Sprite.hpp>
+#include <SFML/Graphics/RectangleShape.hpp>
+#include <SFML/Graphics/RenderTarget.hpp>
+#include <list>
+#include <memory>
+
+class QuadTree : public sf::Drawable {
+    static const unsigned int maxLevel = 15;
+    unsigned int level;
+
+    unsigned int nodeCapacity;
+
+    std::list<sf::Sprite *> nodes;
+
+    bool isSplit;
+
+    std::unique_ptr<QuadTree> northWest;
+    std::unique_ptr<QuadTree> northEast;
+    std::unique_ptr<QuadTree> southWest;
+    std::unique_ptr<QuadTree> southEast;
+
+    sf::RectangleShape shape;
+
+    QuadTree(const QuadTree&) = delete;
+    void operator=(const QuadTree&) = delete;
+public:
+    QuadTree(const unsigned int &x = 0,
+             const unsigned int &y = 0,
+             const unsigned int &width = 1,
+             const unsigned int &height = 1,
+             const unsigned int &node_capacity_ = 5,
+             const unsigned int &level_ = 0);
+
+    void set_NodeCapacity(unsigned int n);
```

Comment:
It was to avoid the reading "setNode Capacity", but I agree this is not a good idea.